cmd/satellite/main.go

```diff
@@ -264,6 +284,39 @@ func cmdQDiag(cmd *cobra.Command, args []string) (err error) {
 	return w.Flush()
 }
 
+func cmdPayments(cmd *cobra.Command, args []string) (err error) {
+	ctx := process.Ctx(cmd)
+
+	layout := "2006-01-02"
+	start, err := time.Parse(layout, args[0])
+	if err != nil {
+		return errs.New("Invalid date format. Please use YYYY-MM-DD")
+	}
+	end, err := time.Parse(layout, args[1])
+	if err != nil {
+		return errs.New("Invalid date format. Please use YYYY-MM-DD")
+	}
+
+	// Ensure that start date is not after end date
+	if start.After(end) {
+		return errs.New("Invalid time period (%v) - (%v)", start, end)
+	}
+
+	output := os.Stdout
+	if out != "" {
+		file, err := os.Create(out)
+		if err != nil {
+			return err
+		}
+		output = file
+	}
+	defer func() {
+		err = errs.Combine(err, output.Close())
```

Comment:
This should be specific to file only, we don't want to close stdout, it would cause problems.